File: GUI/seminar/src/app/models/CatalogItem.ts
import { Catalog } from "./Catalog";
import { CatalogItemImages } from "./catalogItemImages";

export class CatalogItem {
    id: number;
    name: string;
    catalogId: number = 0;
    catalogName: string ="";
    catalog: Catalog;
    description: string = "";
    companyId: number = 0;
    companyName: string = "";
    isEnabled!: boolean;
    catalogItemTypeId: number = 0;
    catalogItemStatusId: number = 0;
    lamella: string;
    floor: number;
    units: number;


    created!: Date;
    createdUser: number = 0;
    modified!: Date;
    modifiedUser: number = 0;

    versionIdentifier: string = "";
    floors: number; 
    rooms: number;
    bathrooms: number;
    garagePlaces: number;
    dimensions: string = "";
    netArea: number; 
    bruttoArea: number; 
    isFeatured!: boolean;

    catalogItemImages: CatalogItemImages[] = []; 

    disclaimer: string;
    isDisclamer: boolean;
    approvedUser: string;
    deploymentStatusId: number = 0;

    constructor() {
        this.id = 0;
        this.name = '';

        this.catalog = new Catalog();
        this.companyName = '';
        this.description = '';
        this.floors = 0;
        this.bathrooms = 0;
        this.rooms = 0;
        this.garagePlaces = 0;
        this.netArea = 0;
        this.bruttoArea = 0;
        this.floor = 0;
        this.lamella = '';
        this.units = 0;
        this.disclaimer = '';
        this.isDisclamer = false;
        this.approvedUser = '';
    }
}

File: API/domain/Entities/realestate/CatalogItem.cs
using domain.Dto.relestate;
using domain.Entities.common;
using domain.Entities.realestate;

namespace domain.Entities.releastate
{
    public class CatalogItem : BaseEntity
    {
        public CatalogItem()
        {
            CatalogItemImages = new HashSet<CatalogItemImage>();
        }
        //Database Atributes
        public int CatalogId { get; set; }
        public string? Name { get; set; }
        public Guid VersionIdentifier { get; set; }
        public string? Quote { get; set; }
        public string? Description { get; set; }
        public float? Floors { get; set; } //Broj katova kod kuće
        public int? Rooms { get; set; }
        public int? Bathrooms { get; set; }
        public int? GaragePlaces { get; set; }
        public string? Lamella { get; set; }
        public int? Floor { get; set; } //Broj kat u zgradi na kojem je stan
        public int? Units { get; set; } //Broj etaza u stanu

        public string? Dimensions { get; set; }
        public float? NetArea { get; set; }
        public float? BruttoArea { get; set; }
        public bool? IsEnabled { get; set; }
        public bool? IsFeatured { get; set; }

        public int? CatalogItemTypeId { get; set; }
        public int? CatalogItemStatusId { get; set; }


        public string? Disclaimer { get; set; }
        public bool? IsDisclamer { get; set; }
        public string? ApprovedUser { get; set; }
        public int? DeploymentStatusId { get; set; }



        //Referent tables
        public virtual Catalog? Catalog { get; set; }
        public virtual CatalogItemType? CatalogItemType { get; set; }
        public virtual CatalogItemStatus? CatalogItemStatus { get; set; }
        public virtual DeploymentStatus? DeploymentStatus { get; set; }

        public virtual ICollection<CatalogItemImage> CatalogItemImages { get; set; } = new List<CatalogItemImage>();

    }
}
